web/src/components/IdeViewer/IdeViewer.js

```diff
@@ -1,25 +1,133 @@
-import { useContext } from 'react'
 import { IdeContext } from 'src/components/IdeToolbarNew'
+import { useRef, useState, useEffect, useContext, useMemo } from 'react'
```

Comment:
This file is becoming pretty coupled to openscad, but I'm in the "get it working" stage. I'll try and seperate concerns after.